### PR TITLE
Correct values for HighMemoryLoadThresholdBytes, MemoryLoadBytes.

### PR DESCRIPTION
**Pull request by naricc**
Correct values fort HighMemoryLoadThresholdBytes, MemoryLoadBytes. I looked in the code for CoreCLR and spoke with Maoni Stephens and Jan Kotas about the values used in CoreCLR, so I am confident they are all correct now.

HighMemoryThresholdBytes is a system wide memory usage threshold at which CoreCLR changes its compaction strategy. There is no direct correspondance to anything in mono-sgen. But since it used by software caches to determine when memory in the system is "nearly" exhauste, mono-sgent will use .9 * total physical memory as an acceptable value.

### CODE REVIEW

File: src/mono/mono/utils/memfuncs.c
Comment (BrzVlad) [inline src/mono/mono/utils/memfuncs.c]: Didn't we copy this code from somewhere ? Shouldn't we remove it from there ?

File: src/mono/mono/sgen/sgen-memory-governor.c
Comment (BrzVlad) [inline src/mono/mono/sgen/sgen-memory-governor.c]: there are 3 typos in this 2 line comment :(

**Comment by BrzVlad** [comment]
@naricc was there something left to fix for TotalAvailableMemoryBytes ? Since that was the name of the old PR

**Comment by naricc** [comment]
> @naricc was there something left to fix for TotalAvailableMemoryBytes ? Since that was the name of the old PR

I had thought that needed fixing, but after looking at CoreCLR, it turns out it does not. It is supposed to be the total amount of physical memory in the machine, regardless if any part of it is already used.  This is the value it already has in mono-sgen.

File: src/mono/mono/utils/memfuncs.c
Comment (CoffeeFlux) [inline src/mono/mono/utils/memfuncs.c]: I guess we're okay using the 32-bit version of this here since the returned value is in pages rather than bytes?

File: src/mono/mono/utils/memfuncs.c
Comment (CoffeeFlux) [inline src/mono/mono/utils/memfuncs.c]: Can't we try `/proc/meminfo` first?

File: src/mono/mono/sgen/sgen-memory-governor.c
Comment (CoffeeFlux) [inline src/mono/mono/sgen/sgen-memory-governor.c]:
```suggestion
		// In sgen it is not adjusted dynamically, since sgen does not adjust compaction strategies based on a threshold.
```

**Review by CoffeeFlux — CHANGES_REQUESTED** [review]
Isn't all this copy-pasted? Please put the function in a header and have it only declared once.

File: src/mono/mono/utils/memfuncs.c
Comment (CoffeeFlux) [reply src/mono/mono/utils/memfuncs.c]: Judging by https://github.com/dotnet/runtime/issues/13371 the answer is yes, we should.

**Review by BrzVlad — APPROVED** [review]

File: src/mono/mono/utils/memfuncs.c
Comment (naricc) [reply src/mono/mono/utils/memfuncs.c]: As this PR fixes a number of issues and is ready to go, I am going to merge it, and create another issue to fix that.

Comment (naricc) [reply src/mono/mono/utils/memfuncs.c]: Since this fixes a number of issues and is ready to, I am incline to merge it and create a seperate issue for this. It is going to take me a little bit to add parsing for /proc and make sure it is consistent on different linuxes.

Comment (CoffeeFlux) [reply src/mono/mono/utils/memfuncs.c]: Sure, though i'd like for it to not fall off the radar since the exact issue has already been hashed out in CoreCLR. It shouldn't be too complicated, since if MemAvailable isn't present in `/proc/meminfo` you can just fall back to sysconf. CoreCLR's implementation, for reference: https://github.com/dotnet/runtime/blob/490551c221be0683039cebc1e5da5b12b3527c12/src/coreclr/src/gc/unix/gcenv.unix.cpp#L823-L854

File: src/mono/mono/utils/memfuncs.c
Comment (naricc) [reply src/mono/mono/utils/memfuncs.c]: I believe that is correct.

**Review by CoffeeFlux — APPROVED** [review]
LGTM, though if this fixes the GCMemoryInfo corefx test please re-enable that before merging and mark this PR as fixing the corresponding issue.

**Comment by naricc** [comment]
@CoffeeFlux It seems there is still an issue with fragmentation regarding that test I will have to look into before I can re-enable it.

**Comment by mikernet** [comment]
Which platforms are affected by the issue this addresses and what is the nature of the issue exactly - is it completely broken, slightly off, etc..?

Are the values provided in .NET Core 3.1 running on Windows reliable for the purpose of cache management?

**Comment by CoffeeFlux** [comment]
@mikernet this is a Mono-specific change, so it does not affect .NET Core 3.1. We only just recently added the API to Mono as part of larger .NET 5 work, so it has yet to ship to anyone yet :)

**Pull request merged**
